File: codes/ruby/inheritance/single/gradstudent.rb
require_relative "student"
require_relative "person"

class GradStudent < Student

    alias_method :parent_info, :info

    def initialize(name, grade)
        super(name, grade)

        puts "======================"
        parent_info # calling info of the parent class
        info # calling info of the current class
        Person.instance_method(:info).bind(self).call
    end

    def info
        puts "GradStudent info"
    end

end

File: codes/php/inheritance/single/gradstudent.php
<?php

require_once "student.php";

class GradStudent extends Student {

    private $grade;

    public function __construct($name, $grade) {
        parent::__construct($name, $grade);
        $this->grade = $grade + 2;
        //parent::parent::info(); //cannot do this
        parent::info();
        //echo parent::luckyNumber;
    }

}

?>
